src/pages/groupComparison/GroupComparisonMutationMapper.tsx

```diff
@@ -46,14 +51,101 @@ export default class GroupComparisonMutationMapper extends MutationMapper<
     }
 
     /**
-     * Override the parent method to get custom controls.
+     * Overriding the parent method to have a customized filter panel.
      */
-    protected get customControls(): JSX.Element | undefined {
+    protected get mutationFilterPanel(): JSX.Element | null {
         return (
-            <PercentToggle
-                axisMode={this.props.axisMode}
-                onScaleToggle={this.props.onScaleToggle}
-            />
+            <div>
+                <div
+                    style={{
+                        paddingTop: 5,
+                        paddingBottom: 5,
+                    }}
+                >
+                    <div style={{ fontWeight: 'bold', paddingBottom: 3 }}>
+                        (A){' '}
+                        {
+                            this.store.dataStore.sortedFilteredGroupedData[0]
+                                .group
+                        }
+                    </div>
+                    <DriverAnnotationProteinImpactTypeBadgeSelector
+                        filter={this.proteinImpactTypeFilter}
+                        counts={this.mutationCountsByProteinImpactTypeForGroup(
+                            0
+                        )}
+                        onSelect={this.onProteinImpactTypeSelect}
+                        onClickSettingMenu={this.props.onClickSettingMenu}
+                        annotatedProteinImpactTypeFilter={
+                            this.annotatedProteinImpactTypeFilter
+                        }
+                    />
+                    <hr style={{ marginTop: 10, marginBottom: 10 }}></hr>
+                    <div style={{ fontWeight: 'bold', paddingBottom: 3 }}>
+                        (B){' '}
+                        {
+                            this.store.dataStore.sortedFilteredGroupedData[1]
+                                .group
+                        }
+                    </div>
+                    <DriverAnnotationProteinImpactTypeBadgeSelector
+                        filter={this.proteinImpactTypeFilter}
+                        counts={this.mutationCountsByProteinImpactTypeForGroup(
+                            1
+                        )}
+                        onSelect={this.onProteinImpactTypeSelect}
+                        onClickSettingMenu={this.props.onClickSettingMenu}
+                        annotatedProteinImpactTypeFilter={
+                            this.annotatedProteinImpactTypeFilter
+                        }
+                    />
+                </div>
+            </div>
+        );
+    }
+
+    protected mutationsGroupedByProteinImpactTypeForGroup(groupIndex: number) {
```

Comment:
if at all possible we should pull this logic apart.   you must already by filtering by drivers somewhere else right?  so, could we not apply this impact type filtering on the result of that?  instead of applying again?

Reply:
i.e. the two other types of filtering we are applying inside this method are not implied by the methods name so it would be easy to misunderstand what it's doing.

Reply:
This is mostly duplicate of https://github.com/cBioPortal/cbioportal-frontend/blob/634421d48da5156540b77993d011a194132339d3/src/shared/components/mutationMapper/MutationMapper.tsx#L466

Actually, we need to apply filters here again by excluding the protein impact type filters. Otherwise we lose the counts for unchecked types since they're filtered out: https://github.com/cBioPortal/cbioportal-frontend/blob/634421d48da5156540b77993d011a194132339d3/src/shared/components/mutationMapper/MutationMapper.tsx#L474-L475

So, we need this additional filtering logic here to dynamically calculate the numbers (inside the boxes) to display for each type. I think we should separate that logic into a utility function or something instead of duplicating the existing code in each new implementation.

Reply:
Actually it's not working properly for this implementation. When a type is deselected, its count immediately drops to zero.

![image](https://user-images.githubusercontent.com/3604198/211105123-eaf5faa2-d54d-410c-b139-ae7ba15ef3bf.png)

We should keep the original count just like in the current production

![image](https://user-images.githubusercontent.com/3604198/211105754-4ab1ebd8-0fef-499c-b3ae-0cf8798084e5.png)

Reply:
This seems fixed now. Thanks!